test/test-debuglet.ts

```diff
@@ -924,6 +924,34 @@ describe('Debuglet', function() {
       debuglet.start();
     });
 
+    it('should resolve promises before exiting user functions', function(done) {
```

Comment:
I don't think there is any relationship between the timing of the promise resolution and exiting a user function that we can make or that we need.  Instead we only need to ensure a breakpoint fetch (and necessary state updates) have occurred before the `Promise` is resolved.

Reply:
Done.